vispy/visuals/volume.py

```diff
@@ -411,7 +411,10 @@
         vec3 max_loc_tex = vec3(0.0);  // Location where the maximum value was encountered
         """,
     in_loop="""
-        sumval = sumval + val;
+        // Scale accumulation in `sumval` by contrast limits so that:
+        // * attenuation value does not depend on data values
+        // * negative values do not amplify instead of attenuate
+        sumval = sumval + (val - clim.x) / (clim.y - clim.x);
```

Comment:
A thought just occurred to me that depending on how `clim` is set this could still result in accumulating negative values. Perhaps this should be changed further to something like this:
```suggestion
        sumval = sumval + max(0.0, (val - clim.x) / (clim.y - clim.x));
```
or even:
```suggestion
        sumval = sumval + min(max(0.0, (val - clim.x) / (clim.y - clim.x)), clim.y);
```

Reply:
Mhm, good point. I think it makes sense? Btw, you can use `clamp()` to do the same as `min(max())`

Reply:
Thanks! I am not too experienced with GLSL. I don't think it's a super-important change (doesn't have an effect with default clim) but it seems worth making so I just pushed another commit to add it.